src/internal/OpenApiTools/components/Parameter.ts

```diff
@@ -50,10 +50,11 @@ export const generatePropertySignatureObject = (
     if (reference.type === "local") {
       context.setReferenceHandler(currentPoint, reference);
       const localRef = store.getParameter(reference.path);
+      const isPathProperty = localRef.in === "path";
       const name = converterContext.escapePropertySignatureName(localRef.name);
       const typeElement = factory.PropertySignature.create({
         name: name,
-        optional: false,
+        optional: isPathProperty ? false : !localRef.required,
```

Comment:
👍 👍 👍 

https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#parameterObject

> Determines whether this parameter is mandatory. If the parameter location is "path", this property is REQUIRED and its value MUST be true. Otherwise, the property MAY be included and its default value is false.